scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,113 +58,155 @@ def generate_load_table_files(dataset_tables_dict):
+            exit()
+
+        # TODO: determine how to handle drops of failed rows/columns to get cleaned data
+        validated_dataset[table] = dataset[table]
+        print(f"Success: Table {table} has been validated.")
+    
+    print(f"Success: Dataset has been validated.")
+    return validated_dataset
 
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
 
-    # if any errors, print error message and exit
-    if errors_index_rows:
-        print(f"Failed: Dataset validation errors found. Please retry after correcting errors listed in validation_errors.csv.")
-        pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
-        exit()
+def col_is_in_defined_fields(col_name, field_dict):
+    list_of_fields = list(field_dict.keys())
+    if col_name in list_of_fields:
+        return True
 
-    # TODO: determine how to handle drops of failed rows/columns to get cleaned data
-    # errors_index_rows = [e.row for e in errors]
-    # data_clean = dataset_metadata_df.drop(index=errors_index_rows, axis=1)
-    # data_clean.to_csv('clean_data.csv')
-    print(f"Success: Dataset has been validated.")
-    return dataset
+    return False
 
 
-def load_excel_input(excel, expected_dataset_cols):
-    """Read excel file input into a dataframe and validate for required columns."""
+def load_excel_input(excel, allowed_dataset_cols, skiprows):
+    """
+    Read excel file input into a dataframe and validate for required columns.
+        Skipping first two rows, as the input excel has two rows of headers before the data starts.
+        Usecols uses a lambda function to filter out columns that aren't defined in the schema.json file's 
+            fields definition section. This is used instead of a list of expected columns, because providing
+            a list enforces that exactly those columns must exist in the data frame, and we don't want to
+            enforce the schema until we're checking it during validation. This should solely be data import.
+    """
 
-    try:
-        dataset_metadata_df = pd.read_excel(excel, sheet_name="Sheet1", skiprows=2, usecols=expected_dataset_cols, index_col=None)
-
-    except ValueError as e:
-        # if the value error is about missing columns in input file
-        if len(e.args) > 0 and e.args[0].startswith("Usecols do not match columns, columns expected but not found"):
-            missing_cols = e.args[0].split(":")[1]
-            print(f"Failed: Input excel file has the following missing columns that are required: {missing_cols}")
-        else:
-            print(e)
-        exit()
-    
+    raw_dataset_df = pd.read_excel(excel, sheet_name=None, skiprows=skiprows, index_col=None, usecols=lambda x: x in allowed_dataset_cols)
```

Comment:
as discussed on Slack: consider adding an "FYI" comment here to notify the user of any columns being dropped at ingest, due to not being in the schema

Reply:
per slack:
So I can't figure out an elegant solution for notifying the user that some columns were dropped. The read_excel API doesn't seem to have an option.
I'm not sure if there's a way to rope that in as a side effect of the lambda function that's evaluating if the column should be included or not.
I could just print the data frames as they exist after import, allowing the user to look over them and see if anything is missing that they expect should be there.
Would that be a reasonable solution if we can't figure something better out?

Reply:
What if you just read in all the data here (i.e. remove usecols limitation) and in the validation you highlight the extra fields present?

Do you know from the user whether they'd prefer to keep extra data or drop it?

Reply:
I talked to the user about handling extra data, but upon discussing with John it was determined to be scope creep and out of scope for this. 

I'm just not sure that this warrants getting rid of the out-of-the-box way of filtering columns on ingest for the read_excel API. I added a notification that pops up when that code is run to notify the user that any columns not present in the schema will be ignored.